strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/users/security/AuthoritiesProvider.java

```diff
@@ -39,88 +32,37 @@
     private AuthorizationConfigFileManager authorizationConfigFileManager;
 
     @PostConstruct
-    void init()
+    void init() throws IOException
     {
         final AuthorizationConfigDto config = authorizationConfigFileManager.read();
         authorizationConfigService.setAuthorizationConfig(config);
     }
 
-    public Set<GrantedAuthority> getAuthoritiesByRoleName(final String roleName)
+    public Set<RoleData> getAssignableRoles()
     {
-        Set<GrantedAuthority> fullAuthorities = new HashSet<>();
-        Set<Role> configuredRoles = new HashSet<>();
-        populate(fullAuthorities, configuredRoles);
-
-        Set<GrantedAuthority> authorities = new HashSet<>();
-
-        // set by SecurityConfig#anonymousAuthenticationFilter()
-        if (AuthorizationConfigService.ANONYMOUS_ROLE.equals(roleName))
-        {
-            authorities.add(new SimpleGrantedAuthority(Privileges.ANONYMOUS_USER.getAuthority()));
-        }
-        else
-        {
-            authorities.add(new SimpleGrantedAuthority(Privileges.AUTHENTICATED_USER.getAuthority()));
-        }
-
-        if (Roles.ADMIN.name().equals(roleName))
-        {
-            authorities.addAll(fullAuthorities);
-        }
-
-        // add all privileges from etc/conf/strongbox-authorization.yaml for any role that defines there
-        configuredRoles.forEach(role ->
-                                {
-                                    if (role.getName()
-                                            .equalsIgnoreCase(roleName))
-                                    {
-                                        role.getPrivileges()
-                                            .forEach(
-                                                    privilegeName -> authorities.add(
-                                                            new SimpleGrantedAuthority(privilegeName.toUpperCase())));
-                                    }
-                                });
-
-        try
+        return authorizationConfigService.get().getRoles();
+    }
+    
+    public Role getRuntimeRole(String name)
+    {
+        RoleData role = authorizationConfigService.get()
+                                              .getRoles()
+                                              .stream()
+                                              .filter(r -> r.getName().equals(name))
+                                              .findFirst()
+                                              .orElseThrow(() -> new IllegalArgumentException(name));
```

Comment:
Cosmetic: Could you please, re-align the chained method calls? Thanks! :)

Reply:
fixed